docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,640 @@
+{
+    ConnectCallback = async (context, cancellationToken) =>
+    {
+        var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+        try
+        {
+            // Setting TCP Keep Alive
+            socket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.KeepAlive, true);
+            socket.SetSocketOption(SocketOptionLevel.Tcp, SocketOptionName.TcpKeepAliveTime, 60);
+            socket.SetSocketOption(SocketOptionLevel.Tcp, SocketOptionName.TcpKeepAliveInterval, 1);
+
+            // Setting ReceiveBufferSize
+            socket.ReceiveBufferSize = 8192;
+
+            // Enabling ReusePort
+            socket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReuseUnicastPort, true);
+
+            // Disabling Nagle Algorithm
+            socket.NoDelay = true;
+
+            await socket.ConnectAsync(context.DnsEndPoint, cancellationToken);
+        }
+        catch
+        {
+            socket.Dispose();
+            throw;
+        }
+        return new NetworkStream(socket, ownsSocket: true);
+    }
+};
+var client = new HttpClient(handler);
+using var response = await client.GetAsync(uri);
+```
+
+### Example: Enabling Dns Round Robin
+
+DNS Round Robin is a technique used to distribute network traffic across multiple servers by rotating through a list of IP addresses associated with a single domain name. This helps in load balancing and improving the availability of services. When using HttpClient, you can implement DNS Round Robin by manually handling the DNS resolution and rotating through the IP addresses using the ConnectCallback property of SocketsHttpHandler.
+
+By enabling DNS Round Robin, you can ensure that your application distributes requests evenly across multiple servers, reducing the load on any single server and improving overall performance and reliability. This approach is particularly useful for high-traffic applications that require efficient load balancing.
+
+To enable DNS Round Robin with HttpClient, you can use the ConnectCallback property to manually resolve the DNS entries and rotate through the IP addresses. Here's an example for `HttpWebRequest` and `HttpClient`:
+
+**Old Code Using `HttpWebRequest`**:
+
+```csharp
+ServicePointManager.DnsRefreshTimeout = 60000;
+ServicePointManager.EnableDnsRoundRobin = true;
+HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
+using HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+```
+
+In the older `HttpWebRequest` API, enabling DNS Round Robin was straightforward due to its built-in support for this feature. However, the newer `HttpClient` API does not provide the same built-in functionality. Despite this, you can achieve similar behavior by implementing a `DnsRoundRobinConnector` that manually rotates through the IP addresses returned by DNS resolution.
+
+**New Code Using `HttpClient`**:
+
+:::code source="../snippets/httpclient/DnsRoundRobin.cs" id="DnsRoundRobinConnector":::
```

Comment:
I still find it odd we've decided to copy an entire project into a sample rather than linking to the NuGet package/source

Reply:
We might also reference to NuGet package, I guess and keep the sample here.